src/main/scala/firrtl/annotations/Target.scala

```diff
@@ -111,6 +120,29 @@ object Target {
   def isOnly(seq: Seq[TargetToken], keywords:String*): Boolean = {
     seq.map(_.is(keywords:_*)).foldLeft(false)(_ || _) && keywords.nonEmpty
   }
+
+  /** @return [[Target]] from human-readable serialization */
+  def deserialize(s: String): Target = {
+    val regex = """(?=[~|>/:.\[@])"""
+    s.split(regex).foldLeft(GenericTarget(None, None, Nil)) { (t, tokenString) =>
+      val value = tokenString.tail
+      tokenString(0) match {
+        case '~' if t.circuitOpt.isEmpty && t.moduleOpt.isEmpty && t.tokens.isEmpty =>
+          if(value == "???") t else t.copy(circuitOpt = Some(value))
+        case '|' if t.moduleOpt.isEmpty && t.tokens.isEmpty =>
+          if(value == "???") t else t.copy(moduleOpt = Some(value))
+        case '/' => t.add(Instance(value))
+        case ':' => t.add(OfModule(value))
+        case '>' => t.add(Ref(value))
+        case '.' => t.add(Field(value))
+        case '[' if value.dropRight(1).toInt >= 0 => t.add(Index(value.dropRight(1).toInt))
+        case '@' if value == "clock" => t.add(Clock)
+        case '@' if value == "init" => t.add(Init)
+        case '@' if value == "reset" => t.add(Reset)
```

Comment:
This one is probably okay, but I think we should benchmark to be sure:

`.add` involves appending to Lists which could get expensive for deeply nested aggregates. For anything < 5 it's fast because those Lists are optimized.

I don't know if this is worth "fixing" without benchmarks showing it's a problem though because it might be kind of a pain to make fast.

Reply:
I changed tokens from a Seq to a Vector, so that should do it